server/gfmd/user.c

```diff
@@ -319,11 +327,11 @@ user_lookup_auth_id_including_invalid(
 
 struct user *
 user_lookup_auth_id(
-	enum auth_user_id_type auth_user_id_type,
+	const char *auth_user_id_type_str,
 	const char *auth_user_id)
 {
 	struct user *u = user_lookup_auth_id_including_invalid(
-			auth_user_id_type, auth_user_id);
+			auth_user_id_type_str, auth_user_id);
```

Comment:
calling gfarm_auth_user_id_type_from_name() here is a waste of CPU time.
instead of changing the first parameter of user_lookup_auth_id_including_invalid() from enum auth_user_id_type to const char *, please keep the parameter type and let the caller directly pass enum auth_user_id_type, by exporting enum auth_user_id_type in user.h.